### PR TITLE
Add else to Erlang's reserved keywords

### LINKED ISSUE #3304: "Else" type variant causes erlang troubles
It looks like I hit an edge case. 

```rs
pub type Example {
  Else
} 
```

I don't know if this will minimally repoduce the problem, but erlang doesn't seem to like using the "else" token.

### PR DESCRIPTION
**Pull request by giacomocavalieri**
This PR closes #3304

### CODE REVIEW

File: compiler-core/src/erlang/tests.rs [hunk 1]
```diff
@@ -632,3 +632,19 @@ pub fn main() {
 "
     );
 }
+
+// https://github.com/gleam-lang/gleam/issues/3304
+#[test]
+fn type_named_else() {
```
Comment (lpil) [inline compiler-core/src/erlang/tests.rs:638]: Another test for functions too please

File: CHANGELOG.md [hunk 2]
```diff
@@ -132,6 +132,10 @@
   JavaScript.
   ([Louis Pilfold](https://github.com/lpil))
 
+- Fixed a bug where calling a type or a constuctor `Else` would generate invalid
+  erlang code.
+  ([Giacomo Cavalieri](https://github.com/giacomocavalieri))
```
Comment (lpil) [inline CHANGELOG.md:137]: Could you merge this with the `maybe` note above please 🙏

**Review by lpil — COMMENTED** [review]
Thank you!

File: compiler-core/src/erlang/tests.rs (hunk 1 above)
Comment (giacomocavalieri) [reply compiler-core/src/erlang/tests.rs:638]: `else` is a keyword in gleam as well so we can't have a function named like that

File: CHANGELOG.md (hunk 2 above)
Comment (giacomocavalieri) [reply CHANGELOG.md:137]: Ok!

File: compiler-core/src/erlang/tests.rs (hunk 1 above)
Comment (lpil) [reply compiler-core/src/erlang/tests.rs:638]: oh yes!!!

**Review by lpil — APPROVED** [review]
Thank you again!

**Pull request merged**